tests/integration/conftest.py

```diff
@@ -512,21 +635,48 @@ def get_branch():
     branch_ref.delete()
 
 
-@pytest_asyncio.fixture(scope="module", name="app_with_grafana_agent")
-async def app_with_grafana_agent_integrated_fixture(
-    model: Model, app_no_runner: Application
+@pytest_asyncio.fixture(scope="module", name="app_for_metric")
+async def app_for_metric_fixture(
+    model: Model,
+    basic_app: Application,
+    instance_type: InstanceType,
+    existing_app: Optional[str],
 ) -> AsyncIterator[Application]:
-    """Setup the charm to be integrated with grafana-agent using the cos-agent integration."""
-    # Pin grafana agent revisions since the recent releases are not compatible with juju 2.9
-    # (due to usage of juju secrets)
-    grafana_agent = await model.deploy(
-        "grafana-agent",
-        channel="latest/edge",
-        revision=105,
-        series="jammy",
-    )
-    await model.relate(f"{app_no_runner.name}:cos-agent", f"{grafana_agent.name}:cos-agent")
-    await model.wait_for_idle(apps=[app_no_runner.name], status=ACTIVE)
-    await model.wait_for_idle(apps=[grafana_agent.name])
+    # OpenStack integration does not need the grafana agent to collect metric.
+    if instance_type == InstanceType.LOCAL_LXD and not existing_app:
+        grafana_agent = await model.deploy(
+            "grafana-agent",
+            application_name=f"grafana-agent-{basic_app.name}",
+            channel="latest/edge",
+        )
```

Comment:
Note that there is a bug for subordinates and architectures: https://bugs.launchpad.net/juju/+bug/2067749
This may fail in the future if the downloaded charm does not match the architecture. So maybe pinning a specific rev would be good.
We might also consider changing the lxd mode to always collect and output metrics, so we can skip this deployment here.

Reply:
Good suggestion.
I think this would not be a part of the openstack integration.
I think we should track this in a separate story.

Reply:
Yes, you might want to pin the revision to the latest current revision for amd (I assume we are using amd runners here for the test). We are not using the grafana agent in the tests here anyway, so having the latest revision is not critical.